File: io-hotmoka-network/src/main/java/io/hotmoka/network/internal/models/transactions/InitializationTransactionRequestModel.java
package io.hotmoka.network.internal.models.transactions;

import io.hotmoka.network.internal.models.storage.StorageReferenceModel;

public class InitializationTransactionRequestModel extends InitialTransactionRequestModel {
    private StorageReferenceModel manifest;
    private String classpath;

    public String getClasspath() {
        return classpath;
    }

    public void setClasspath(String classpath) {
        this.classpath = classpath;
    }

    public StorageReferenceModel getManifest() {
        return manifest;
    }

    public void setManifest(StorageReferenceModel manifest) {
        this.manifest = manifest;
    }
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-js/src/models/requests/InitializationTransactionRequestModel.ts
import {StorageReferenceModel} from "../values/StorageReferenceModel";
import {TransactionReferenceModel} from "../values/TransactionReferenceModel";
import {InitialTransactionRequestModel} from "./InitialTransactionRequestModel";
import {HotmokaException} from "../../internal/exception/HotmokaException";


export class InitializationTransactionRequestModel extends InitialTransactionRequestModel {
    /**
     * The reference to the jar containing the basic Takamaka classes. This must
     * have been already installed by a previous transaction.
     */
    manifest: StorageReferenceModel

    /**
     * The storage reference that must be set as manifest.
     */
    classpath: TransactionReferenceModel

    constructor(manifest: StorageReferenceModel, classpath: TransactionReferenceModel) {
        super()

        if (classpath === null || classpath === undefined) {
            throw new HotmokaException("classpath cannot be null or undefined")
        }

        if (manifest === null || manifest === undefined) {
            throw new HotmokaException("manifest cannot be null or undefined")
        }

        this.manifest = manifest
        this.classpath = classpath
    }
}